integration_tests/src/main/python/conditionals_test.py

```diff
@@ -97,6 +99,34 @@ def test_case_when(data_gen):
                 f.when(f.lit(False), f.col('_c0'))),
             conf = allow_negative_scale_of_decimal_conf)
 
+# Seperate this test to avoid OOM in premerge build.
+@pytest.mark.parametrize('data_gen', all_gens + all_nested_gens + decimal_128_gens, ids=idfn)
+@pytest.mark.parametrize('pred_value', [True, False, None])
+def test_case_when_all_true_false(data_gen, pred_value):
+    # Smaller number to reduce the memory size for this big test. Bigger number may get
+    # OOM easily and be killed in premerge build.
```

Comment:
The fact that we had to write a separate test to avoid an OOM screams this is not something we want to do purely in search of a performance win.

Reply:
Updated.

Reply:
Merged the two tests